src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -837,9 +837,38 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
     auto m = std::make_shared<pattern::Matcher>(gather, matcher_name);
     register_matcher(m, matcher_pass_callback);
 }
+ov::pass::EliminateSlice::EliminateSlice() {
+    MATCHER_SCOPE(EliminateSlice);
 
-ov::pass::NopStridedSlice::NopStridedSlice() {
-    MATCHER_SCOPE(NopStridedSlice);
+    auto input = pattern::any_input();
+    auto begin_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto end_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto step_const = pattern::wrap_type<ov::op::v0::Constant>();
+    auto axes = pattern::any_input();
+    auto pattern = pattern::wrap_type<ov::op::v8::Slice>({input, begin_const, end_const, step_const, axes});
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        auto slice = std::dynamic_pointer_cast<ov::op::v8::Slice>(m.get_match_root());
+
+        int64_t max_int = slice->input_value(2).get_element_type() == element::i32
```

Comment:
only element::i32? what about another i[0-9]+ and u[0-9]+? i64, u32, ...

Reply:
no, both i32 and i64 are supported.